util/chunk/list.go

```diff
@@ -71,6 +71,16 @@ func (l *List) NumChunks() int {
 	return len(l.chunks)
 }
 
+// FieldTypes returns the fieldTypes of the list
+func (l *List) FieldTypes() []*types.FieldType {
+	return l.fieldTypes
+}
+
+// NumRowsOfChunk returns the number of rows of a chunk in the ListInDisk.
+func (l *List) NumRowsOfChunk(chkID int) int {
+	return l.chunks[chkID].NumRows()
+}
```

Comment:
This method seems redundant since we can use `List.GetChunk(chkID).NumRows()`.

Reply:
You are right, however `ListInDisk` implemented NumRowsOfChunk and can't be removed because there is no `ListInDisk.GetChunk`, therefore I prefer to keep it for the sake of isometry. 

```
// NumRowsOfChunk returns the number of rows of a chunk in the ListInDisk.
func (c *RowContainer) NumRowsOfChunk(chkID int) int {
	if c.tmp != nil && chkID == c.NumChunks()-1 {
		return c.tmp.NumRows()
	}
	if c.AlreadySpilled() {
		return c.recordsInDisk.NumRowsOfChunk(chkID)
	}
	return c.records.NumRowsOfChunk(chkID)
}
```